File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{

    float change = 0;
    int dollar = 0;
    int coins = 0;

    // prompt input then convert dollars to cents
    do
    {
        change = get_float("Change owed: ");
    }
    while (change <= 0);

    dollar = round(change * 100);

    // subtracting with quarters
    while (dollar >= 25)
    {
        dollar = dollar - 25;
        coins++;
    }

    // subtracting with dimes
    while (dollar >= 10 && dollar < 25)
    {
        dollar = dollar - 10;
        coins++;
    }

    // subtracting with nickels
    while (dollar >= 5 && dollar < 10)
    {
        dollar = dollar - 5;
        coins++;
    }

    // subtracting with pennies
    while (dollar >= 1 && dollar < 5)
    {
        dollar = dollar - 1;
        coins++;
    }

    // print final amount of coins
    {
        printf("%i\n", coins);
    }

}

File: pset6/cash/cash.py
from cs50 import get_float
coins = 0

while True:
    change = get_float("Changed owed: ")
    if change <= 0:
        change = get_float("Changed owed: ")
    if change > 0:
        break


dollar = round(change * 100)

while dollar >= 25:
    dollar = dollar - 25
    coins += 1
while dollar >= 10 and dollar < 25:
    dollar = dollar - 10
    coins += 1
while dollar >= 5 and dollar < 10:
    dollar = dollar - 5
    coins += 1
while dollar >= 1 and dollar < 5:
    dollar = dollar - 1
    coins += 1

print(f"{coins}")